galley/pkg/runtime/processor.go

```diff
@@ -49,38 +52,58 @@ type Processor struct {
 	stopped chan struct{}
 
 	// The current in-memory configuration State
-	state *State
+	state         *State
+	stateStrategy *publish.Strategy
+
+	distributor publish.Distributor
 
 	// hook that gets called after each event processing. Useful for testing.
 	postProcessHook postProcessHookFn
 
 	// lastEventTime records the last time an event was received.
 	lastEventTime time.Time
+
+	fullSyncCond *sync.Cond
```

Comment:
Nit: Please add a comment explaining the conditions usage model.

Reply:
done.